[NEW] Support sending audio records

luismachado opened the pull request:
@RocketChat/ios

I forked #762 (as no update was given since a month ago)  so credit also to @falcaoaugustos (maybe it's best to update his PR with this one?)

Changes since the fork:
- Visual feedback is different now (whole screen is blurred while recording and a timer is also present)
- Feedback is now quicker as I replaced the button action for a tapgesturerecognizer (`delayTouchesBegan` was also removed)
- Code is also up to date with the most recent commits.

rafaelks commented:
@luismachado Looks like this one have conflicts? 🤔

luismachado commented:
@rafaelks Yeah also noticed that. Looking into it now.

luismachado commented:
@rafaelks should be without conflicts now. Let me know what you think :)

rafaelks commented on Rocket.Chat.xcodeproj/project.pbxproj:
```diff
@@ -1691,7 +1700,7 @@
 					41DF76DE1D2C50710028DBF8 = {
 						CreatedOnToolsVersion = 7.3.1;
-						DevelopmentTeam = S6UPZG7ZR3;
+						DevelopmentTeam = MJHBLUH3VK;
```
Can you remove this change?

rafaelks commented on Rocket.Chat.xcodeproj/project.pbxproj:
```diff
@@ -2543,12 +2555,12 @@
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
-				DEVELOPMENT_TEAM = S6UPZG7ZR3;
+				DEVELOPMENT_TEAM = MJHBLUH3VK;
```
Can you remove this change?

rafaelks commented on Rocket.Chat.xcodeproj/project.pbxproj:
```diff
@@ -2543,12 +2555,12 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
 				OTHER_SWIFT_FLAGS = "$(inherited) -D DEBUG \"-D\" \"COCOAPODS\"";
-				PRODUCT_BUNDLE_IDENTIFIER = chat.rocket.ios;
+				PRODUCT_BUNDLE_IDENTIFIER = chat.rocket.ios11;
```
We also need to remove this change.

rafaelks commented on Rocket.Chat.xcodeproj/project.pbxproj:
```diff
@@ -2565,11 +2577,11 @@
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
-				DEVELOPMENT_TEAM = S6UPZG7ZR3;
+				DEVELOPMENT_TEAM = MJHBLUH3VK;
```
Can you remove this change?

rafaelks commented on Rocket.Chat.xcodeproj/project.pbxproj:
```diff
@@ -2565,11 +2577,11 @@
 				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = chat.rocket.ios;
+				PRODUCT_BUNDLE_IDENTIFIER = chat.rocket.ios11;
```
Can you remove this change?

rafaelks commented on Rocket.Chat/en.lproj/Localizable.strings:
```diff
@@ -28,6 +28,13 @@
+
+"alert.audio_message.error.title" = "Ooops!";
+"alert.audio_message.error.message" = "An error has occurred. Please, record your message again.";
+
```
We've to add this strings to the other localizable files (in English is fine if you don't know the other languages). Is just to keep it in sync.

luismachado commented:
@rafaelks Okay, just addressed your remarks. Regarding the translation, I was going to add the PT version but it looks like it was already there :P 

Therefore I just added the lines for the other 3 but left a TODO tag near each one, as I don't trust google translate with longer phrases...

rafaelks commented:
Looking good, but I noticed something curious... always, the first time I send the audio, it doesn't work, stays with 00:00 after the upload. Can you also reproduce this issue?

luismachado commented:
Yeah, also just noticed that. It's also happening in @falcaoaugustos branch. 
It looks to always be the first recording per launch (even if you don't send the 1st one the 2nd one will be okay).

Going to try to find out what's happening.

luismachado commented:
Okay, the issue seems now seems to be fixed. 

Would you care to check it on your end @rafaelks ? Thanks.

rafaelks commented:
Thank you @luismachado & @falcaoaugustos!!! 🚀

falcaoaugustos commented:
Thank you very much @luismachado . I had a few problems fixing it but I will read your code in order to learn how to do so. I must say that I am still studying this case.
Thank you too, @rafaelks . By the way, I will make more contributions as soon as I solve my money issue.
